exporter/signalfxexporter/exporter.go

```diff
@@ -60,7 +60,6 @@ type signalfxExporter struct {
 	hostMetadataSyncer *hostmetadata.Syncer
 	converter          *translation.MetricsConverter
 	dimClient          *dimensions.DimensionClient
-	cancelFn           func()
```

Comment:
The cancelFn was originally responsible for shutting down all sub-packages. I've refactored to enable each package to start and shutdown their own functionality, as requested by the caller. (More explanation in PR description).